### PR TITLE
Correctly use `NamedTempFile`

### LINKED ISSUE #1723: diesel_cli v1.3 print-schema broken on WIN10
## Setup
With diesel_cli 1.2:

`diesel print-schema` gives correct output.

With diesel_cli 1.3:
`diesel print-schema` gives 
> Access is denied. (os error 5)

### Versions

- **Rust:** nightly-2018-05-14/1.26 Stable
- **Diesel:** 1.3
- **Database:** PG10
- **Operating System** WIN10

### Feature Flags

- **diesel:** Postgres

## Checklist

- [ X ] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.
- [ X ] This issue can be reproduced on Rust's stable channel.

Doing some troubleshooting, but it's 1 am and I might pass out. Will figure out cause tomorrow at latest.

### PR DESCRIPTION
**Pull request by weiznich**
Do not try to open the generated temp file twice. This will fail on
windows.

Fixes #1723

### CODE REVIEW

**Comment by weiznich** [comment]
@Deedasmi Could you check if that solves the issue? (I've no windows machine 😉 )

**Comment by Deedasmi** [comment]
Confirmed it works! Thanks!

File: diesel_cli/src/print_schema.rs [hunk 1]
```diff
@@ -40,21 +40,22 @@ pub fn run_print_schema(
     config: &config::PrintSchema,
 ) -> Result<(), Box<Error>> {
     let tempfile = NamedTempFile::new()?;
-    output_schema(database_url, config, tempfile.path())?;
+    let file = tempfile.reopen()?;
+    output_schema(database_url, config, file, tempfile.path())?;
 
     // patch "replaces" our tempfile, meaning the old handle
     // does not include the patched output.
-    let mut file = File::open(tempfile.path())?;
```
Comment (sgrif) [inline diesel_cli/src/print_schema.rs]: I think this line needs to remain unchanged.

**Comment by weiznich** [comment]
@Deedasmi (or someone other with access to a windows machine) Could you try again? The old solution had some issues. (Hopefully now everything works fine.

**Comment by sgrif** [comment]
@weiznich You can push this to the `appveyor_nonsense` branch to get an appveyor run, too

**Comment by sgrif** [comment]
@Deedasmi The problem is that we end up giving the path to `patch`, which will replace the file not just truncate/append, so our old handle becomes invalid.

**Comment by Deedasmi** [comment]
Good call, edited. Trying to set up tests/patch test now.

**Comment by sgrif** [comment]
@weiznich Can you release this as 1.3.1?

**Comment by sgrif** [comment]
Waiting on CI

**Pull request merged**